ratik/src/main/java/org/opencadc/ratik/InventoryValidator.java

```diff
@@ -150,7 +169,215 @@ public InventoryValidator(Map<String, Object> daoConfig, URI resourceID, Artifac
+                message = "local before remote";
+            } else {
+                message = "local after remote";
+            }
+            log.debug("order: " + message);
+        }
+        return result;
+    }
+
+    /**
+     * Get local artifacts matching the uriBuckets.
+     *
+     * @param bucket The bucket prefix.
+     * @return ResourceIterator over Artifact's matching the remote filter policy and the uri buckets.
+     *
+     * @throws ResourceNotFoundException For any missing required configuration that is missing.
+     * @throws IOException               For unreadable configuration files.
+     */
+    ResourceIterator<Artifact> getLocalIterator(final String bucket)
+        throws ResourceNotFoundException, IOException {
+        String constraint = null;
+        if (StringUtil.hasText(this.artifactSelector.getConstraint())) {
+            constraint = this.artifactSelector.getConstraint().trim();
+        }
+        // order query results by Artifact.uri
+        boolean ordered = true;
+        return this.artifactDAO.iterator(constraint, bucket, ordered);
+    }
+
+    /**
+     * Execute the query and return the iterator back.
+     *
+     * @param bucket The bucket prefix.
+     * @return ResourceIterator over Artifact's matching the remote filter policy and the uri buckets.
+     *
+     * @throws ResourceNotFoundException For any missing required configuration that is missing.
+     * @throws IOException               For unreadable configuration files.
+     * @throws IllegalStateException     For any invalid configuration.
+     * @throws TransientException        temporary failure of TAP service: same call could work in future
+     * @throws InterruptedException      thread interrupted
+     */
+    ResourceIterator<Artifact> getRemoteIterator(final String bucket)
+        throws ResourceNotFoundException, IOException, IllegalStateException, TransientException, InterruptedException {
+        final TapClient<Artifact> tapClient = new TapClient<>(this.resourceID);
+        final String query = buildRemoteQuery(bucket);
+        log.debug(String.format("\nExecuting query '%s'\n", query));
+        return tapClient.execute(query, new ArtifactRowMapper());
+    }
+
+    /**
+     * Assemble the WHERE clause and return the full query.  Very useful for testing separately.
+     *
+     * @param bucket The current bucket.
+     * @return  String query.  Never null.
+     */
+    String buildRemoteQuery(final String bucket)
+        throws ResourceNotFoundException, IOException {
+        final StringBuilder query = new StringBuilder();
+        query.append("SELECT id, uri, contentChecksum, contentLastModified, contentLength, contentType, ")
+            .append("contentEncoding, lastModified, metaChecksum FROM inventory.Artifact ");
+
+        if (StringUtil.hasText(this.artifactSelector.getConstraint())) {
+            log.debug(String.format("\nInjecting artifact selector clause '%s'\n",
```

Comment:
rather than these two debug messages just log the entire query somewhere at debug level; that is always more useful. I would be more inclined to put that in getRemoteIterator... oh wait: it is already there... so these debug statements here are just noise.

Reply:
Removed the two debug messages.